packages/account/src/providers/provider.ts

```diff
@@ -379,15 +379,28 @@ type ChainInfoCache = Record<string, ChainInfo>;
  */
 type NodeInfoCache = Record<string, NodeInfo>;
 
-type Operations = ReturnType<typeof getOperationsSdk>;
-
-type SdkOperations = Omit<Operations, 'submitAndAwait' | 'statusChange'> & {
-  submitAndAwait: (
-    ...args: Parameters<Operations['submitAndAwait']>
-  ) => Promise<ReturnType<Operations['submitAndAwait']>>;
-  statusChange: (
-    ...args: Parameters<Operations['statusChange']>
-  ) => Promise<ReturnType<Operations['statusChange']>>;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type PromisifyReturn<OfFn extends (...args: any) => any> = (
+  ...args: Parameters<OfFn>
+) => Promise<ReturnType<OfFn>>;
+
+type Ops = ReturnType<typeof getOperationsSdk>;
+
+type Operations = {
+  [K in keyof Ops]: ReturnType<Ops[K]> extends AsyncIterable<unknown>
+    ? PromisifyReturn<Ops[K]> // Wrap subscriptions in a promise to match return of `FuelGraphqlSubscriber`
+    : Ops[K];
+};
+
+type SdkOperations = Omit<Operations, 'submitAndAwait'> & {
+  /**
+   * This method is DEPRECATED and will be REMOVED in v1.
+   *
+   * This method will hang until the transaction is fully processed, as described in https://github.com/FuelLabs/fuel-core/issues/2108.
+   *
+   * Please use the `submitAndAwaitStatus` method instead.
+   */
+  submitAndAwait: PromisifyReturn<Operations['submitAndAwait']>;
```

Comment:
The [repetition](https://github.com/FuelLabs/fuels-ts/pull/3101/files#diff-69a6670d0426c74935ed2830ce552e7df2379c2d112a582306e57569fcd17029L385) was bothering me so I created a helper.

Reply:
Abstraction is a double-edged sword that should never be favored when it adds complexity to no apparent real benefit. Moving back and forth and translating what's going on with the helper is way more complicated than just reading the former alternative [repetition]—which, by the way, appeared only once and is not even repeating. Did I miss something?

Anyway, because we read code several times more than we write code, simplicity and objectivity are always preferred from a readability and maintenance point of view.

Reply:
I added the abstraction to force-wrap all subscriptions, existing and future ones, so that they match the runtime return type of `FuelGraphqlSubscriber` as explained in the [code comment](https://github.com/FuelLabs/fuels-ts/pull/3101/files#diff-69a6670d0426c74935ed2830ce552e7df2379c2d112a582306e57569fcd17029R391).

Alas, I have reverted the change in https://github.com/FuelLabs/fuels-ts/pull/3101/commits/b56889a7a6be02cc856f57d33df20f9671543054.